libs/langchain-community/src/vectorstores/elasticsearch.ts

```diff
@@ -342,15 +343,27 @@ export class ElasticVectorSearch extends VectorStore {
             [metadataField]: condition.value,
           },
         });
-      } else {
+      } else if (condition.operator === "or") {
+        should.push({
+          term: {
+            [metadataField]: condition.value,
+          },
+        });
+      }else {
         must.push({
           [condition.operator]: {
             [metadataField]: condition.value,
           },
         });
       }
     }
-    return { must, must_not };
+    const result = { must, must_not };
+    
+    if(should.length > 0){
```

Comment:
can you also extend the fn return type to include should please?

Reply:
Done.